src/query/api/v1/handler/prometheus/native/read_common.go

```diff
@@ -191,16 +196,19 @@ func read(
 
 	// Detect clients closing connections.
 	if cancelWatcher != nil {
-		ctx, cancel := context.WithTimeout(ctx, fetchOpts.Timeout)
+		var cancel context.CancelFunc
+		ctx, cancel = context.WithTimeout(ctx, parsed.FetchOpts.Timeout)
 		defer cancel()
-
 		cancelWatcher.WatchForCancel(ctx, cancel)
 	}
 
 	bl, err := engine.ExecuteExpr(ctx, parser, opts, fetchOpts, params)
 	if err != nil {
 		return emptyResult, err
 	}
+	if err := ctx.Err(); err != nil {
```

Comment:
this shouldn't be needed, since the error should bubble up from `ExecuteExpr`

Reply:
I had added just for testing purposes since I wasn't easily able to mock `ExecuteExpr` to respect the timeout. But that isn't really a good reason haha - I'll actually mock this properly